File: pset1/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int countChangeCoins(int n);
int main()
{
    int change;
    // Ask the change to the user until the condition is satisfied
    do
    {
        change = get_float("Change Owed: ");
    }
    while (change <= 0);

    // Rounding the decimals and *100
    change = round(change);

    // Calling the function that calculates the number of coins
    int coins = countChangeCoins(change);

    printf("%i\n", coins);

    return 0;
}

int countChangeCoins(int n)
{
    int coins = 0;

    do
    {
        if (n >= 25)
        {
            n = n - 25;
            coins++;
        }
        else if (n >= 10)
        {
            n = n - 10;
            coins++;
        }
        else if (n >= 5)
        {
            n = n - 5;
            coins++;
        }
        else
        {
            n = n - 1;
            coins++;
        }
    }
    while (n > 0);

    return coins;
}

File: pset6/cash.py
def main():
    change = get_float("Change: ")
    change = round(change * 100)
    coins = countChangeCoins(change)
    print(coins)


def get_float(prompt):
    while True:
        try:
            value = float(input(prompt))
            if value > 0:
                return value
            else:
                pass
        except:
            pass


def countChangeCoins(value):
    coins = 0
    while value > 0:
        if value >= 25:
            value = value - 25
            coins += 1
        elif value >= 10:
            value = value - 10
            coins += 1
        elif value >= 5:
            value = value - 5
            coins += 1
        elif value >= 1:
            value = value - 1
            coins += 1
    return coins


main()
